### PR TITLE
add preloadProvider - for preloading data.

### PR DESCRIPTION
**Pull request by rhutchison**
I had a need to preload data to my angular app. After some research, I came up with this.

Credit: http://bitwit.ca/blog/regarding-angularjs-data-preloading/

### CODE REVIEW

File: app/views/layout.server.view.html [hunk 1]
```diff
@@ -40,6 +40,13 @@
 	<!--[if lt IE 9]>
 		<script src="http://html5shim.googlecode.com/svn/trunk/html5.js"></script>
 	<![endif]-->
+
+	{% if user %}
+	<!--Embedding The User Object-->
+	<script type="text/preloaded" name="user">
```
Comment (ilanbiala) [inline app/views/layout.server.view.html]: What is the `name` attribute for?

File: public/application.js [hunk 2]
```diff
@@ -4,11 +4,24 @@
...
-]);
+angular.module(ApplicationConfiguration.applicationModuleName).config(['$locationProvider', function($locationProvider) {
+	$locationProvider.hashPrefix('!');
+}])
+.constant('$preloadProvider', {})
+.config(['$preloadProvider', function($preloadProvider) {
+    var scripts = document.getElementsByTagName('script');
+    for (var i = 0; i < scripts.length; i++) {
+        var script = scripts[i];
+        if (script.type === 'text/preloaded') {
```
Comment (ilanbiala) [inline public/application.js]: This part seems way to easy to break if someone changes a single attribute. Any other ideas on how to handle it?

File: app/views/layout.server.view.html (hunk 1 above)
Comment (rhutchison) [inline app/views/layout.server.view.html]: @ilanbiala, see below:
* name is not a valid attribute on a script tag.
* name serves as the array key of $preloadProvider constant.

File: public/application.js (hunk 2 above)
Comment (rhutchison) [inline public/application.js]: What is the concern?  There are only two supported attributes, type & name.

**Comment by ilanbiala** [comment]
@lirantal what do you think about this? It definitely needs to wait until 0.4.0 is released, but is this something we should consider?

**Comment by lirantal** [comment]
@ilanbiala  @rhutchison  I don't like all the for loop handling there to parse the DOM, and having to neat pick data from an element's HTML value.

@rhutchison can you come up with other ways to work around the problem you are describing? there must be a more elegant way to do this.

**Comment by rhutchison** [comment]
I found it to be pretty elegant.  How is this any different then how in-line templates are handled?

**Comment by ilanbiala** [comment]
I don't like the for loop either, and I don't think many people will need the extra complexity.

**Comment by lirantal** [comment]
@rhutchison I'll close this PR since I believe there is a more proper way of implementing this, and also this PR is against 'master' branch and not 0.4.0

We'll get back to it later if you wish, with a newer PR.

**Pull request closed without merging**